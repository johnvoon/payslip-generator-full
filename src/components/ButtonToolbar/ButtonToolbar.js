import React from "react";
import PropTypes from "prop-types";
import Button from "components/Button";

const ButtonToolbar = ({ pristine, submitting, reset, handleSubmit }) => {
  return (
    <div className="btn-toolbar">
      <Button
        customClassNames="btn-danger"
        type="button"
        disabled={pristine || submitting}
        handleClick={reset}
      >
        Reset Field
      </Button>
      <Button
        customClassNames="btn-primary"
        type="submit"
        disabled={submitting}
        handleClick={handleSubmit}
      >
        Generate Payslip Data
      </Button>
    </div>
  );
};

ButtonToolbar.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  pristine: PropTypes.bool.isRequired,
  reset: PropTypes.func.isRequired,
  submitting: PropTypes.bool.isRequired
};

export default ButtonToolbar;
