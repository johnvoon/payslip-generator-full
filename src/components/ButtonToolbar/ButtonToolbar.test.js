import React from "react";
import ButtonToolbar from "./ButtonToolbar";
import { shallow } from "enzyme";

describe("ButtonToolbar", () => {
  const reset = jest.fn();
  const handleSubmit = jest.fn();

  const renderedComponent = shallow(
    <ButtonToolbar
      pristine={false}
      submitting={false}
      reset={reset}
      handleSubmit={handleSubmit}
    />
  );

  it("should match snapshot", () => {
    expect(renderedComponent).toMatchSnapshot();
  });
});
