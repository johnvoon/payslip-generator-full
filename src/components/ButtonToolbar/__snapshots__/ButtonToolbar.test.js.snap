exports[`ButtonToolbar should match snapshot 1`] = `
<div
  className="btn-toolbar">
  <Button
    customClassNames="btn-danger"
    disabled={false}
    handleClick={[Function]}
    type="button">
    Reset Field
  </Button>
  <Button
    customClassNames="btn-primary"
    disabled={false}
    handleClick={[Function]}
    type="submit">
    Generate Payslip Data
  </Button>
</div>
`;
